scripts/globals/job_utils/rune_fencer.lua

```diff
@@ -50,8 +50,163 @@ local function enforceRuneCounts(target)
     end
 end
 
+--source https://www.bluegartr.com/threads/124844-Vivacious-Pulse-testing?p=6038534&viewfull=1#post6038534 and following posts
+local function getRuneHealAmount(effect, target)
+
+    if effect == xi.effect.TENEBRAE then -- Tenebrae primarily restores MP, but also uses the base divine skill/2 bonus but no raw stat bonus.
+        return 0
+    end
+
+    local runeStatMap = {
+                            [xi.effect.IGNIS]  = xi.mod.STR,
+                            [xi.effect.GELUS]  = xi.mod.DEX,
+                            [xi.effect.FLABRA] = xi.mod.VIT,
+                            [xi.effect.TELLUS] = xi.mod.AGI,
+                            [xi.effect.SULPOR] = xi.mod.INT,
+                            [xi.effect.UNDA]   = xi.mod.MND,
+                            [xi.effect.LUX]    = xi.mod.CHR,
+                        }
+
+    local stat = runeStatMap[effect:getType()]
```

Comment:
If the below is possible to be nil, this will throw an exception and break out of the function if getType does not exist in the above table.

Reply:
I adjusted this logic so that nil check is no longer necessary. Take a look!